core/src/memory.rs

```diff
@@ -152,6 +152,31 @@ impl Memory {
 		Ok(())
 	}
 
+	/// Copy memory region form `src` to `dst` with length.
+	/// `copy_within` used `memmove` to avoid DoS attacks.
+	pub fn copy(
+		&mut self,
+		src_offset: usize,
+		dst_offset: usize,
+		length: usize,
```

Comment:
Is this function supposed to be a no-op if `length == 0` similar to the other memory functions? If so then it should have an early exit condition at the top of the function similar to what `copy_large` has.

Reply:
Actually, this case if covered in `mcopy` function. But I added extra check for it.